src/common/function/src/scalars/json/json_path_exists.rs

```diff
@@ -64,32 +78,61 @@ impl Function for JsonPathExistsFunction {
         let paths = &columns[1];
 
         let size = jsons.len();
-        let datatype = jsons.data_type();
         let mut results = BooleanVectorBuilder::with_capacity(size);
 
-        match datatype {
-            // JSON data type uses binary vector
-            ConcreteDataType::Binary(_) => {
+        match (jsons.data_type(), paths.data_type()) {
+            (ConcreteDataType::Binary(_), ConcreteDataType::String(_)) => {
                 for i in 0..size {
                     let json = jsons.get_ref(i);
                     let path = paths.get_ref(i);
 
-                    let json = json.as_binary();
-                    let path = path.as_string();
-                    let result = match (json, path) {
-                        (Ok(Some(json)), Ok(Some(path))) => {
-                            let json_path = jsonb::jsonpath::parse_json_path(path.as_bytes());
-                            match json_path {
-                                Ok(json_path) => jsonb::path_exists(json, json_path).ok(),
-                                Err(_) => None,
-                            }
+                    let json = if let Ok(Some(bytes)) = json.as_binary() {
+                        bytes
+                    } else {
+                        // This should never happen.
+                        return InvalidFuncArgsSnafu {
+                            err_msg: format!("Illegal json binary: {:?}", json),
                         }
-                        _ => None,
+                        .fail();
+                    };
+
+                    let result = if !path.is_null() {
+                        // Since path is not null, we simply
+                        // unwrap it without checking if it's `Null`.
+                        let path = if let Ok(Some(str)) = path.as_string() {
+                            if let Ok(json_path) = jsonb::jsonpath::parse_json_path(str.as_bytes())
+                            {
+                                json_path
+                            } else {
+                                // Should never happen.
+                                return InvalidFuncArgsSnafu {
+                                    err_msg: format!("Illegal json path {:?}", path),
+                                }
+                                .fail();
+                            }
+                        } else {
+                            // Should never happen.
+                            return InvalidFuncArgsSnafu {
+                                err_msg: format!("Illegal json path: {:?}", path),
+                            }
+                            .fail();
+                        };
+
+                        jsonb::path_exists(json, path).ok()
+                    } else {
+                        // Should never reach here since we `match` in advance.
+                        None
                     };
```

Comment:
nit: can we simplify this logic

Reply:
Sure. I tried simplifying, please take a look. Let me know if there's anything more.